### PR TITLE
Add "Expand" example to Haptic Sampler

### PR DESCRIPTION
**Pull request by jmylen**

### CODE REVIEW

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        fun provideFactory(
+            application: Application,
+        ): ViewModelProvider.Factory = object : ViewModelProvider.Factory {
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                val vibrator = application.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
+
+                var messageToUser = ""
+                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R || !vibrator.areAllPrimitivesSupported(
```
Comment (laissandrade) [inline Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandViewModel.kt:47]: Ins't this < R?

**Review by laissandrade — CHANGES_REQUESTED** [review]
Thanks! Just waiting to confirm the updated primitive scale and timings before approving.

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandViewModel.kt (hunk 1 above)
Comment (jmylen) [reply Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandViewModel.kt]: Good catch

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S &&
+        VIBRATION_DATA_FOR_EXPANDING.allPrimitivesSupported(vibrator) &&
+        VIBRATION_DATA_FOR_COLLAPSING.allPrimitivesSupported(vibrator)
+    ) {
+        animateToExpandedDuration = remember {
+            VIBRATION_DATA_FOR_EXPANDING.getDuration(vibrator)
+        }
+        animateToCollapsedDuration = remember {
+            VIBRATION_DATA_FOR_COLLAPSING.getDuration(vibrator)
+        }
```
Comment (laissandrade) [inline Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt:126]: Please consider passing on the default values to the getDuration methods, so you don't need to do API and primitive checks here, you can delegate to the PrimitiveComposition object.

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    }
+
+    val thickness by transition.animateFloat(
+        transitionSpec = {
+            when {
+                ExpandShapeState.Collapsed isTransitioningTo ExpandShapeState.Expanded ->
+                    tween(durationMillis = animateToExpandedDuration, easing = LinearEasing)
+                else ->
+                    tween(durationMillis = animateToCollapsedDuration, easing = LinearEasing)
+            }
```
Comment (laissandrade) [inline Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt:254]: Could you please extract and reuse this, like you did for the Resist example?

**Review by laissandrade — APPROVED** [review]
Thanks!

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt (hunk 2 above)
Comment (jmylen) [reply Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt]: good call

File: Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt (hunk 3 above)
Comment (jmylen) [reply Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt]: done

**Pull request merged**